src/Analyzers/CSharp/CodeFixes/UseThrowExpression/UseThrowExpressionCodeFixProvider.cs

```diff
@@ -49,17 +52,31 @@ public override Task RegisterCodeFixesAsync(CodeFixContext context)
 
             foreach (var diagnostic in diagnostics)
             {
-                var ifStatement = root.FindNode(diagnostic.AdditionalLocations[0].SourceSpan);
+                var startNode = root.FindNode(diagnostic.AdditionalLocations[0].SourceSpan);
                 var throwStatementExpression = root.FindNode(diagnostic.AdditionalLocations[1].SourceSpan);
                 var assignmentValue = root.FindNode(diagnostic.AdditionalLocations[2].SourceSpan);
+                var expressionStatement = assignmentValue.GetAncestor<ExpressionStatementSyntax>();
 
-                // First, remote the if-statement entirely.
+                var ifStatement = startNode is IfStatementSyntax ifStatementSyntax ? ifStatementSyntax : (IfStatementSyntax)startNode.ChildNodes().First(node => node is IfStatementSyntax);
+                var closeBrace = ifStatement.CloseParenToken;
+                var block = ifStatement.Statement is BlockSyntax blockSyntax ? blockSyntax.Statements.First() : ifStatement.Statement;
+
+                var triviaList = new SyntaxTriviaList()
+                    .AddRange(closeBrace.GetAllTrailingTrivia().Where(t => !t.IsWhitespaceOrEndOfLine()))
+                    .AddRange(block.GetLeadingTrivia().Where(t => !t.IsWhitespaceOrEndOfLine()))
+                    .AddRange(block.GetTrailingTrivia().Where(t => !t.IsWhitespaceOrEndOfLine()))
+                    .Add(SyntaxFactory.ElasticCarriageReturnLineFeed);
+
+                // First, remove the if-statement entirely.
                 editor.RemoveNode(ifStatement);
 
-                // Now, update the assignment value to go from 'a' to 'a ?? throw ...'.
-                editor.ReplaceNode(assignmentValue,
-                    generator.CoalesceExpression(assignmentValue,
-                    generator.ThrowExpression(throwStatementExpression)));
+                // Now, update the assignment value to go from 'a' to 'a ?? throw 
+                var assignment = expressionStatement.Expression as AssignmentExpressionSyntax;
```

Comment:
this should probably become just a regular cast to `AssignmentExpressionSyntax`, I think CI will fail with a nullable warning here right now